File: backend/arkham-core/library/Arkham/Types/SkillType.hs
module Arkham.Types.SkillType where

import Arkham.Prelude

newtype CommittedSkillIcon = CommittedSkillIcon { unCommittedSkillIcon :: SkillType }
  deriving newtype (Show, Eq, Generic, Ord, ToJSON, FromJSON)

data SkillType
  = SkillWillpower
  | SkillIntellect
  | SkillCombat
  | SkillAgility
  | SkillWild
  deriving stock (Show, Eq, Generic, Ord)
  deriving anyclass (ToJSON, FromJSON)

File: frontend/src/arkham/types/SkillType.ts
import * as JsonDecoder from 'ts.data.json';

export type SkillType = 'SkillWillpower' | 'SkillIntellect' | 'SkillCombat' | 'SkillAgility' | 'SkillWild';

export const skillTypeDecoder = JsonDecoder.oneOf<SkillType>([
  JsonDecoder.literal('SkillWillpower'),
  JsonDecoder.literal('SkillIntellect'),
  JsonDecoder.literal('SkillCombat'),
  JsonDecoder.literal('SkillAgility'),
  JsonDecoder.literal('SkillWild'),
], 'SkillType');
